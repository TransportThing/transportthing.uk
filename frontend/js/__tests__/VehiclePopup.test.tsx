import React from "react";
import renderer from "react-test-renderer";
import { Delay } from "../VehiclePopup";

it("shows delay", () => {
  const component = renderer.create(<Delay item={{ delay: 212 }} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component.update(<Delay item={{ delay: -230 }} />);
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component.update(<Delay item={{ delay: 20 }} />);
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component.update(<Delay item={{ delay: -20 }} />);
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component.update(<Delay item={{}} />);
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
