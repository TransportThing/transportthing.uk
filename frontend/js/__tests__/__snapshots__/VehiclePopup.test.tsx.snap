// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows delay 1`] = `
<div>
  4 minutes late
</div>
`;

exports[`shows delay 2`] = `
<div>
  4 minutes early
</div>
`;

exports[`shows delay 3`] = `
<div>
  On time
</div>
`;

exports[`shows delay 4`] = `
<div>
  On time
</div>
`;

exports[`shows delay 5`] = `null`;
